dlite_entities_service/models/soft.py

```diff
@@ -0,0 +1,148 @@
+"""Base (minimum set) models for SOFT entities."""
+from __future__ import annotations
+
+import difflib
+import re
+from typing import Annotated, Any
+
+from pydantic import (
+    AliasChoices,
+    BaseModel,
+    ConfigDict,
+    Field,
+    field_validator,
+    model_validator,
+)
+from pydantic.networks import AnyHttpUrl
+
+from dlite_entities_service.service.config import CONFIG
+
+URI_REGEX = re.compile(
+    r"^(?P<namespace>https?://.+)/(?P<version>\d(?:\.\d+){0,2})/(?P<name>[^/#?]+)$"
+)
+"""Regular expression to parse a SOFT entity URI."""
+
+
+class SOFTProperty(BaseModel):
+    """The minimum set of defining metadata for a SOFT Entity's property."""
+
+    model_config = ConfigDict(extra="forbid")
+
+    type_: Annotated[
+        str,
+        Field(
+            alias="type",
+            description="The type of the described property, e.g., an integer.",
+        ),
+    ]
+    shape: Annotated[
+        list[str] | None,
+        Field(
+            description=(
```

Comment:
This description seems really complex. Maybe something like <written by chatgpt> "Defines the dimensions of a multi-dimensional property using a list of expressions. "
                "Each expression in the list corresponds to a dimension and can be a named dimension "
                "(e.g., 'H', 'K', 'L') or an arithmetic expression involving these dimensions. "
                "For example, if a property's shape is defined as ['K', 'H+1'] and the dimensions of the entity "
                "are assigned values 'H=2', 'K=2', 'L=6', the resulting shape of the property will be [2, 3]. "
                "This allows for dynamic sizing of properties based on the entity's dimensions. <ChatGPT said this, is it true?>" I'm personally ab it confused by how multi-dimensions work now

Reply:
Technically this comment is not in the scope of the PR since it was already there but I point it out now...

Reply:
Feel free to open an issue on this :)